### PR TITLE
Add parallel=False flag to Group()

### PR DESCRIPTION
**Pull request by gitmarek**
...to allow allocation of parallel groups on supernova.

Ref.: #https://github.com/gitmarek/supriya/commit/cef6ef88d10d5571a038a7bdc1eb2e2ad3b29f78#r87738788

Locally, all tests passed, so it looks like I didn't break anything. Before squashing anything, I'm going to test it manually and wait for supernova to be deployed on GHA to try to write some unit tests too...

### CODE REVIEW

**Comment by josiah-wolf-oberholtzer** [comment]
Honestly, if you have time, tinker with what's going on in https://github.com/josiah-wolf-oberholtzer/supriya/pull/271 and see if you can make supernova run inside GHA. I won't be able to devote much time to that until next week.

**Comment by josiah-wolf-oberholtzer** [comment]
Testing specifics aside, this looks like the implementation I was thinking of.

If we can't get supernova to run inside GHA in a reasonable timeframe, you can always write tests decorated with `pytest.mark.skipif(os.environ.get("CI") == "true")` (or something similar) to skip them inside GHA.

**Comment by gitmarek** [comment]
Tested the code with my local setup.  Seems to work fine.

I run `scsynth` first:

```
scsynth -l 8 -u 12345
```

and the following Python code:
```
import supriya

s = supriya.Server().connect(ip_address="127.0.0.1", port=12345)

s.add_group()
s.add_group(parallel=True)
```

With `sclang`, I query the server (see ref: #https://github.com/gitmarek/supriya/commit/cef6ef88d10d5571a038a7bdc1eb2e2ad3b29f78#r87840094):

```
// sclang: 
s = Server(\test, NetAddr("127.0.0.1", 12345), ServerOptions())
s.initTree;
s.queryAllNodes;
```

and get the following from `scsynth` in the terminal:
```
NODE TREE Group 0
   1 group
      1001 group
      1000 group
   2 group
   3 group
   4 group
   5 group
   6 group
   7 group
   8 group
```

Then I run `supernova`:

```
supernova -u 12345 
```

same Python code and the same `sclang` query, and get:

```
NODE TREE Group 0
   0 group
      1 group
         1001 parallel group
         1000 group
```

**Comment by gitmarek** [comment]
Started to test it in another branch: gitmarek#5.  Will rebase once #274 or similar is merged

**Comment by josiah-wolf-oberholtzer** [comment]
I merged #274 in (thanks!). Go ahead and rebase whenever.

**Comment by josiah-wolf-oberholtzer** [comment]
The pattern I've followed to date is that the low-level "command" classes like `GroupNewRequest` map one-to-one with the published commands in [Server Command Reference](https://doc.sccode.org/Reference/Server-Command-Reference.html). Let's keep it like that and just use the `ParallelGroupNewRequest` as its own command, switching internally. There's more to type in the "backend", but very little of it is user-facing.

For testing, use the OSC-sniffing technique `with server.osc_protocol.capture() as transcript:`. There are some examples of this in the realtime tests: https://github.com/josiah-wolf-oberholtzer/supriya/blob/main/tests/realtime/test_Group.py#L1373-L1411.  It's enough to know that we sent the `/p_new` message to the Server. That's our contract. If `scsynth` or `supernova` doesn't fulfill that contract on their end, that's not our responsibility, so I wouldn't worry (yet) about implementing a usable `GROUP_DUMP_TREE`.

**Comment by josiah-wolf-oberholtzer** [comment]
Also want to say that getting`GROUP_DUMP_TREE` to work is compelling. Testing large node trees with `GROUP_QUERY_TREE` can overflow the 8192 byte packet limit, making it unusable, and I never finished implementing the `/s_info` command for querying a single synth's SynthDef and values.

But it's going to be tricky for a couple of reasons:
- The "sync" Server doesn't continue to read lines from its process output after booting, so we'll have to decide on a mechanism
- There's no "sentinel" value to indicate when the `/g_dumpTree` output finishes, so some kind of timeout mechanism would probably be necessary to prevent blocking forever while waiting for input

File: supriya/realtime/nodes.py
Comment (josiah-wolf-oberholtzer) [inline supriya/realtime/nodes.py:679]: I like the three-line construction you use elsewhere: 

```
request_method = supriya.commands.GroupNewRequest
if self._parallel:
    request_method = supriya.commands.ParallelGroupNewRequest
```

**Review by josiah-wolf-oberholtzer — APPROVED** [review]
LGTM!

Let me know if you're satisfied and I'll merge and cut a new release.

**Comment by gitmarek** [comment]
:+1:

**Comment by josiah-wolf-oberholtzer** [comment]
@gitmarek I'm holding off on releasing until I get OSX builds working again in GHA.

**Pull request merged**
